File: python/BRIDGE_WHOLE_PART/BRIDGE_WHOLE_PART/bridge/Reservacion.py
from BRIDGE_WHOLE_PART.bridge.ProductoCine import ProductoCine
from .ImplementadorProductoCine import ImplementadorProductoCine
from BRIDGE_WHOLE_PART.whole_part.reservacion.ReservacionSencilla import ReservacionSencilla
from BRIDGE_WHOLE_PART.whole_part.reservacion.ReservacionVIP import ReservacionVIP

class Reservacion(ProductoCine):
    
    implementadorProductoCine : ImplementadorProductoCine
    
    def __init__(self, tipoProductoCine, datos) -> None:
        super().__init__(tipoProductoCine)
        
        if self.tipoProductoCine == "sencilla":
            self.implementadorProductoCine = ReservacionSencilla()
        elif self.tipoProductoCine == "vip":
            self.implementadorProductoCine = ReservacionVIP()
        else: print("No existe ese producto!")
        
        self.crearProductoCine(datos)
    
    def crearProductoCine(self, datos):
        self.crearReservacion(datos)
    
    def crearReservacion(self, datos):
        self.implementadorProductoCine.completarProductoCine(datos)
    
    def getReservacion(self):
        return self.implementadorProductoCine
    
    def mostrarReservacion(self):
        self.implementadorProductoCine.mostrarProductoCine()

File: php/BRIDGE_WHOLE_PART/bridge/Reservacion.php
<?php

require_once './bridge/ProductoCine.php';
require_once './whole_part/reservacion/ReservacionSencilla.php';
require_once './whole_part/reservacion/ReservacionVIP.php';


class Reservacion extends ProductoCine{
    
    public $implementadorProductoCine;
    
    public function __construct($tipoProductoCine) {
        parent::__construct($tipoProductoCine);  
    }
    
    public function crearProducto($datos) {
        
         if ($this->tipoProductoCine == 'sencilla'){
            $this->implementadorProductoCine = new ReservacionSencilla();
        }else if($this->tipoProductoCine == 'VIP'){
            $this->implementadorProductoCine = new ReservacionVIP();
        }else{
            echo 'No existe ese producto';
        }
        
        $this->crearReservacion($datos);
    }
    
    public function crearReservacion($datos) {
        $this->implementadorProductoCine->completarProductoCine($datos);
    }
    public function getReservacion() {
        return $this->implementadorProductoCine;
    }
    
    public function mostrarReservacion() {
        return $this->implementadorProductoCine->mostrarProductoCine();
    }


    
}
